src/components/vm/disks/diskAdd.jsx

```diff
@@ -471,20 +473,19 @@ export const AddDiskModalBody = ({ disk, idPrefix, isMediaInsertion, vm, vms, su
     useEffect(() => {
         const file = diskParams.file;
 
-        if (file?.endsWith(".iso")) {
-            setDiskParams(diskParams => ({ ...diskParams, device: "cdrom" }));
-        }
-
         if (file) {
             setVerificationInProgress(true);
             cockpit.spawn(["head", "--bytes=16", file], { binary: true, err: "message", superuser: "try" })
                     .then(file_header => {
-                        let format;
+                        const params = {};
+
+                        if (file.endsWith(".iso"))
+                            params.device = "cdrom";
 
                         // https://git.qemu.org/?p=qemu.git;a=blob;f=docs/interop/qcow2.txt
                         if (file_header[0] == 81 && file_header[1] == 70 &&
                             file_header[2] == 73 && file_header[3] == 251) {
-                            format = "qcow2";
+                            params.format = "qcow2";
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1331-20231128-160227-c60fcb6b-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)